### PR TITLE
[DMS-129] additions

### PR DESCRIPTION
**Pull request by bradbanister**

### CODE REVIEW

File: src/core/EdFi.DataManagementService.Core.External/Frontend/FrontendRequest.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 using EdFi.DataManagementService.Core.External.Model;
 
-namespace EdFi.DataManagementService.Core.Model;
+namespace EdFi.DataManagementService.Core.External.Frontend;
```
Comment (bradbanister) [inline src/core/EdFi.DataManagementService.Core.External/Frontend/FrontendRequest.cs:8]: Not sure why this is wrong but fixed it everywhere

File: src/core/EdFi.DataManagementService.Core.External/Frontend/IFrontendResponse.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
 
-namespace EdFi.DataManagementService.Core.Model;
+namespace EdFi.DataManagementService.Core.External.Frontend;
```
Comment (bradbanister) [inline src/core/EdFi.DataManagementService.Core.External/Frontend/IFrontendResponse.cs:6]: Not sure why this is wrong but fixed it everywhere

File: src/core/EdFi.DataManagementService.Core.Tests.Unit/Middleware/ExtractDocumentInfoMiddlewareTests.cs [hunk 3]
```diff
@@ -60,28 +61,22 @@ public async Task Setup()
...
                 ),
                 RequestMethod.POST
             )
             {
-                ResourceSchema = resourceSchema
+                ResourceSchema = resourceSchema,
+                ParsedBody = JsonNode.Parse(body)!
             };
 
-            if (context.FrontendRequest.Body != null)
```
Comment (bradbanister) [inline src/core/EdFi.DataManagementService.Core.Tests.Unit/Middleware/ExtractDocumentInfoMiddlewareTests.cs:76]: I made this much simpler. You are the owner of the setup, no need to do anything conditionally.

File: src/core/EdFi.DataManagementService.Core.Tests.Unit/Middleware/ValidateDocumentMiddlewareTests.cs [hunk 4]
```diff
@@ -174,7 +174,7 @@ public async Task Setup()
         }
 
         [Test]
-        public void It_should_not_have_response()
+        public void It_should_have_response()
```
Comment (bradbanister) [inline src/core/EdFi.DataManagementService.Core.Tests.Unit/Middleware/ValidateDocumentMiddlewareTests.cs:177]: Updated test name to match

File: src/core/EdFi.DataManagementService.Core/Middleware/ParseBodyMiddleware.cs [hunk 5]
```diff
@@ -22,18 +22,21 @@ public async Task Execute(PipelineContext context, Func<Task> next)
...
                 try
                 {
-                    var body = JsonNode.Parse(context.FrontendRequest.Body);
+                    JsonNode? body = JsonNode.Parse(context.FrontendRequest.Body);
 
-                    if (body != null)
-                    {
-                        context.ParsedBody = body;
-                    }
+                    Trace.Assert(body != null, "Unable to parse JSON");
```
Comment (bradbanister) [inline src/core/EdFi.DataManagementService.Core/Middleware/ParseBodyMiddleware.cs:27]: If `body` is null after `JsonNode.Parse()`, it's a fatal error.

File: src/core/EdFi.DataManagementService.Core/Middleware/ValidateEqualityConstraintMiddleware.cs [hunk 6]
```diff
@@ -15,17 +14,33 @@
...
-internal class ValidateEqualityConstraintMiddleware(ILogger _logger, IEqualityConstraintValidator _equalityConstraintValidator) : IPipelineStep
+internal class ValidateEqualityConstraintMiddleware(
+    ILogger _logger,
+    IEqualityConstraintValidator _equalityConstraintValidator
+) : IPipelineStep
 {
+    /// <summary>
+    /// Use to avoid HTML escaping in output message that we construct.
+    /// </summary>
+    private static readonly JsonSerializerOptions _serializerOptions =
```
Comment (bradbanister) [inline src/core/EdFi.DataManagementService.Core/Middleware/ValidateEqualityConstraintMiddleware.cs:30]: Unrelated optimization I noticed

**Review by JBrenesSimpat — APPROVED** [review]

**Pull request merged**